### ISSUE TITLE
Scan based on string (--github-string)

### ISSUE THREAD

**Issue by Regala** [N0MoreSecr3ts/wraith#114]
Hi!

Wondering if it would be possible to scan based a number of strings. An often example is searching for an internal domain, e.g. `uberinternal.com`. This might be a huge scan, but can definitely pay off.  
(maybe this feature can be triggered already, not sure?)

Thanks!

**Comment by mattyjones** [comment]
You can add it as a regex in the signatures file you pull locally. If it is common, or general enough it can be added to the signatures repo.

**Comment by Regala** [comment]
Hum let me rephrase this then, if you were trying to scan Uber repos for secrets, particularly those who had that `uberinternal.com` how would you do it? 

Thanks!

**Comment by circleous** [comment]
You can use `PartContent` if it's a string inside a file,

```yaml
PatternSignatures:
    -
        part: PartContent
        match: '(?i)uberinternal\.com'
        description: 'Contains uberinternal.com'
        signatureid: '02024b46cfabc3c5b0900fd4cbff6a0b'
        enable: 1
        entropy: 3.5
```

**Comment by BrunoJu** [comment]
@circleous 
Hi pal, I have added your PatternSignatures into file _~/.wraith/signatures/default.yaml_
And then add "uberinternal.com" this string to one of my public repo but wraith can't scan out this string.
Do you know why? 
Thx.

**Comment by BrunoJu** [comment]
> @circleous
> Hi pal, I have added your PatternSignatures into file _~/.wraith/signatures/default.yaml_
> And then add "uberinternal.com" this string to one of my public repo but wraith can't scan out this string.
> Do you know why?
> Thx.

I found I need to change the item value 'commit-depth = 1' up to the number of my commits in config.yml. 
But why need I to change this value manually?  That makes no sense in my view.

**Comment by Regala** [comment]
👆

**Comment by mattyjones** [comment]
@circleous @Regala @BrunoJu The method for scanning here is the correct way at this point. There are some other hacks you can set with the confidence number to only scan for specific signatures. I have thought about implementing a flag that does a single one-off direct scan of a regex. Haven't really sat down to play with it yet though.

The *commit-depth* being set from 1 to another number is explained in #117 a little better (hopefully). The code is in [session.go L253 - L260](https://github.com/N0MoreSecr3ts/wraith/blob/2215c386d81e21075fc4189a3b4e9b678084a8bd/core/session.go#L253-L260)
